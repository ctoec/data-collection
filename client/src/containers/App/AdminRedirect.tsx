import React, { useContext } from 'react';
import UserContext from '../../contexts/UserContext/UserContext';
import { LoadingWrapper } from '@ctoec/component-library';
import { Redirect } from 'react-router-dom';

export const AdminRedirect: React.FC = ({ children }) => {
  const { loading, user } = useContext(UserContext);

  if (loading) {
    return <LoadingWrapper loading={true} />;
  }

  return user?.isAdmin ? <>{children}</> : <Redirect to="/" />;
};