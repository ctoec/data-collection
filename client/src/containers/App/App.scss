nav.usa-nav {
	ul li ul li {
		list-style: none;
	}
}