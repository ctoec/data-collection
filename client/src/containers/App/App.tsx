import React, { useContext } from 'react';
import { Switch } from 'react-router-dom';
import { SimpleHeader } from '@ctoec/component-library';

import routes from '../../routes';
import MakeRouteWithSubRoutes from './MakeRouteWithSubroute';
import UserContext from '../../contexts/UserContext/UserContext';
import { MailToLink } from '../../components/MailToLink';

const App: React.FC = () => {
  const { user } = useContext(UserContext);
  return (
    <div className="App">
      <SimpleHeader
        secondaryTitle="Early childhood enrollment data"
        primaryTitle="Office of Early Childhood"
        loginPath="/login"
        logoutPath="/logout"
        navItems={[
          {
            id: 'templates-nav',
            title: 'Data template',
            children: [
              {
                id: 'excel-template-nav',
                title: 'Excel',
                path: '/upload_template/ECE Data Collection Template.xlsx',
              },
              {
                id: 'csv-template-nav',
                title: '.csv',
                path: '/upload_template/ECE Data Collection Template.csv',
              },
            ],
            renderer: (props) => (
              <a {...props} onClick={undefined} href={props.value}>
                {props.text}
              </a>
            ),
          },
          {
            id: 'data-definitions-nav',
            title: 'Data definitions',
            path: '/column-metadata',
          },
          {
            id: 'feedback-nav',
            title: 'Feedback',
            renderer: () => <MailToLink text="Feedback" />,
          },
        ]}
        userFirstName={user?.firstName}
      />
      <Switch>
        {routes.map((route, index) => (
          <MakeRouteWithSubRoutes key={index} {...route} />
        ))}
      </Switch>
    </div>
  );
};

export default App;
