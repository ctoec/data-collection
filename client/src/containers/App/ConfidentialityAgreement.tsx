import { Alert, Button, Checkbox } from '@ctoec/component-library';
import React, { useContext, useState } from 'react';
import UserContext from '../../contexts/UserContext/UserContext';

export const ConfidentialityAgreement: React.FC = ({ children }) => {
  const {
    confidentialityAgreedDate,
    setConfidentialityAgreedDate,
  } = useContext(UserContext);
  const [_confidentialityAgreed, _setConfidentialityAgreed] = useState(false);

  return !!confidentialityAgreedDate ? (
    <>{children}</>
  ) : (
    <div className="grid-container">
      <Alert
        text="You must read and agree to the privacy policy before using ECE Reporter."
        type="info"
      />
      <div className="grid-row">
        <div className="tablet:grid-col-8">
          <h1>Confidentiality agreement</h1>
          <p>
            Connecticut’s Early Childhood Information System (ECIS) contains
            Personally Identifiable Information (PII) that is confidential
            pursuant to federal and state law. By logging onto and accessing
            ECIS, I hereby acknowledge that the information contained therein
            shall only be disclosed to authorized personnel and used as
            permitted by federal and state law. I further agree that I will only
            use this information for the Connecticut Office of Early Childhood
            conduct of business.
          </p>
          <p>
            I hereby assure that all records, reports and written information
            comply with the following: Provisions of the Family Educational
            Rights and Privacy Act, 20 U.S.C. Section 1232g (FERPA) and the
            regulations promulgated thereunder regarding the disclosure of
            confidential student information
          </p>
          <ol>
            <li>
              Provisions of the Family Educational Rights and Privacy Act, 20
              U.S.C. Section 1232g (FERPA) and the regulations promulgated
              thereunder regarding the disclosure of confidential student
              information;
            </li>
            <li>
              IDEA Part B regulations as identified in 34 CFR §§ 300.610 through
              300.626 and IDEA Part C regulations as identified in 34 CFR §§
              303.400 through 303.417 regarding confidentiality requirements to
              young children with disabilities, and
            </li>
            <li>
              Provisions of the Health Insurance Portability and Accountability
              Act (HIPAA) and the regulations promulgated thereunder.
              Information obtained pursuant to the provisions of HIPAA shall not
              be used or disclosed by the parties for any purpose without
              written consent, or unless otherwise permitted under HIPAA.
            </li>
          </ol>
          <div className="margin-top-4 margin-bottom-4">
            <Checkbox
              id="confidentiality-checkbox"
              text="I have read and agreed to the confidentiality agreement"
              checked={!!confidentialityAgreedDate}
              onChange={(_) => _setConfidentialityAgreed((agreed) => !agreed)}
            />
          </div>
          <div>
            <Button text="Logout" href="/logout" appearance="outline" />
            <Button
              text="Continue to ECE Reporter"
              disabled={!_confidentialityAgreed}
              onClick={() => setConfidentialityAgreedDate(new Date())}
            />
          </div>
        </div>
      </div>
    </div>
  );
};
