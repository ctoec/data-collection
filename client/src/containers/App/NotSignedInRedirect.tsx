import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';
import UserContext from '../../contexts/UserContext/UserContext';

export const NotSignedInRedirect: React.FC = ({ children }) => {
  const history = useHistory();
  const { loading, user } = useContext(UserContext);

  if (loading) {
    return <></>;
  } else {
    if (!user) {
      history.push('/');
      return <></>;
    }
    return <>{children}</>;
  }
};
