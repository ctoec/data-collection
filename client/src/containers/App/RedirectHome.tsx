import React from 'react';
import { Redirect } from 'react-router-dom';

export default function RedirectHome() {
  return <Redirect to="/" />;
}
