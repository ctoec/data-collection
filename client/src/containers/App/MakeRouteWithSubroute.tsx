import React from 'react';
import { Route } from 'react-router-dom';
import { RouteConfig } from '../../routes';
import { ConfidentialityAgreement } from './ConfidentialityAgreement';
import { NotSignedInRedirect } from './NotSignedInRedirect';

// Derived from: https://www.freecodecamp.org/news/hitchhikers-guide-to-react-router-v4-c98c39892399/

const MakeRouteWithSubRoutes = (route: RouteConfig) => {
  return (
    <Route
      path={route.path}
      exact={route.exact}
      render={(props) => {
        const component = (
          <route.component {...props} {...route.props} routes={route.routes} />
        );
        return route.unauthorized ? (
          component
        ) : (
          <NotSignedInRedirect>
            <ConfidentialityAgreement>{component}</ConfidentialityAgreement>
          </NotSignedInRedirect>
        );
      }}
    />
  );
};

export default MakeRouteWithSubRoutes;
