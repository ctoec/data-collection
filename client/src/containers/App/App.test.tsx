import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { BrowserRouter } from 'react-router-dom';

it('matches snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
