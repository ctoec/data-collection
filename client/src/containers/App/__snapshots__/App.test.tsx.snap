// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <a
      class="usa-skipnav"
      href="#main-content"
    >
      Skip to main content
    </a>
    <div
      class=""
    >
      <div
        class="usa-overlay"
        data-testid="overlay"
      />
      <header
        class="usa-header usa-header--extended usa-header--oec-default"
        role="banner"
      >
        <div
          class="usa-navbar"
        >
          <div
            class="usa-logo"
            id="extended-logo"
          >
            <a
              aria-label="ECE Reporter home"
              href="/"
            >
              <div
                class="usa-logo__text display-flex flex-align-center"
              >
                <div>
                  <div
                    class="primary-title primary-title--only"
                  >
                    ECE Reporter
                  </div>
                </div>
              </div>
            </a>
          </div>
          <button
            class="usa-menu-btn"
          >
            Menu
          </button>
        </div>
        <nav
          aria-label="Primary navigation"
          class="usa-nav"
        >
          <div
            class="usa-nav__inner"
          >
            <button
              class="usa-nav__close"
            >
              <img
                alt="close"
              />
            </button>
            <div
              class="usa-nav__secondary usa-nav__secondary--extended"
            >
              <div
                class="usa-nav__secondary-links"
              >
                <nav
                  class="bx--header__nav"
                >
                  <ul
                    class="bx--header__menu-bar"
                  >
                    <li>
                      <a
                        class="bx--header__menu-item"
                        href="/roster"
                        tabindex="0"
                      >
                        <span
                          class="bx--text-truncate--end"
                        >
                          <span
                            class=""
                          >
                            Roster
                          </span>
                        </span>
                      </a>
                    </li>
                    <li
                      class="bx--header__submenu"
                    >
                      <a
                        aria-expanded="false"
                        aria-haspopup="menu"
                        class="bx--header__menu-item bx--header__menu-title"
                        href="#"
                        tabindex="0"
                      >
                        Resources
                        <svg
                          aria-hidden="true"
                          class="bx--header__menu-arrow"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                          />
                        </svg>
                      </a>
                      <ul
                        class="bx--header__menu"
                      >
                        <li>
                          <a
                            class="bx--header__menu-item"
                            href="/template"
                            tabindex="0"
                          >
                            <span
                              class="bx--text-truncate--end"
                            >
                              Data template
                            </span>
                          </a>
                        </li>
                        <li>
                          <a
                            class="bx--header__menu-item"
                            href="/data-requirements"
                            tabindex="0"
                          >
                            <span
                              class="bx--text-truncate--end"
                            >
                              Data requirements
                            </span>
                          </a>
                        </li>
                        <li>
                          <a
                            class="bx--header__menu-item"
                            href="/privacy"
                            tabindex="0"
                          >
                            <span
                              class="bx--text-truncate--end"
                            >
                              Privacy policy
                            </span>
                          </a>
                        </li>
                      </ul>
                    </li>
                    <li
                      class="bx--header__submenu"
                    >
                      <a
                        aria-expanded="false"
                        aria-haspopup="menu"
                        class="bx--header__menu-item bx--header__menu-title"
                        href="#"
                        tabindex="0"
                      >
                        Help
                        <svg
                          aria-hidden="true"
                          class="bx--header__menu-arrow"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                          />
                        </svg>
                      </a>
                      <ul
                        class="bx--header__menu"
                      >
                        <li>
                          <a
                            class="bx--header__menu-item"
                            href="https://help.ece-reporter.ctoec.org/"
                            tabindex="0"
                            target="_blank"
                          >
                            <span
                              class="bx--text-truncate--end"
                            >
                              How-to guides
                            </span>
                          </a>
                        </li>
                        <li>
                          <a
                            class="bx--header__menu-item"
                            href="/support"
                            tabindex="0"
                          >
                            <span
                              class="bx--text-truncate--end"
                            >
                              Support requests
                            </span>
                          </a>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <main
      id="main-content"
    >
      <div
        class="oec-loading-wrapper"
      >
        <div
          aria-atomic="true"
          aria-labelledby="loading-id-1"
          aria-live="assertive"
          class="bx--loading"
        >
          <label
            class="bx--visually-hidden"
            id="loading-id-1"
          >
            Active loading indicator
          </label>
          <svg
            class="bx--loading__svg"
            viewBox="0 0 100 100"
          >
            <title>
              Active loading indicator
            </title>
            <circle
              class="bx--loading__stroke"
              cx="50%"
              cy="50%"
              r="44"
            />
          </svg>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
